memgpt/server/rest_api/openai_assistants/assistants.py

```diff
@@ -0,0 +1,505 @@
+import asyncio
+from fastapi import FastAPI
+from asyncio import AbstractEventLoop
+from enum import Enum
+import json
+import uuid
+from typing import List, Optional, Union
+from datetime import datetime
+
+from fastapi import APIRouter, Depends, Body, HTTPException, Query, Path
+from pydantic import BaseModel, Field, constr, validator
+from starlette.responses import StreamingResponse
+
+from memgpt.server.rest_api.interface import QueuingInterface
+from memgpt.server.server import SyncServer
+
+from memgpt.server.server import SyncServer
+from memgpt.server.rest_api.interface import QueuingInterface
+from memgpt.server.rest_api.static_files import mount_static_files
+from memgpt.models.openai import (
+    AssistantFile,
+    MessageFile,
+    OpenAIAssistant,
+    OpenAIThread,
+    OpenAIMessage,
+    OpenAIRun,
+    OpenAIRunStep,
+    MessageRoleType,
+    Text,
+    ImageFile,
+    ToolCall,
+    ToolCallOutput,
+)
+from memgpt.data_types import LLMConfig, EmbeddingConfig, Message
+from memgpt.constants import DEFAULT_PRESET
+
+"""
+Basic REST API sitting on top of the internal MemGPT python server (SyncServer)
+
+Start the server with:
+  cd memgpt/server/rest_api
+  poetry run uvicorn server:app --reload
+"""
+
+interface: QueuingInterface = QueuingInterface()
+server: SyncServer = SyncServer(default_interface=interface)
+
+
+# router = APIRouter()
+app = FastAPI()
```

Comment:
I think we should probably put this inside `rest_api/server`?

So that when you spin up the memgpt server with `memgpt server`, you get a single FastAPI server that's fielding both the MemGPT API requests + the OpenAI Assistants API requests?

Reply:
This sounds super useful. Looking forward to trying